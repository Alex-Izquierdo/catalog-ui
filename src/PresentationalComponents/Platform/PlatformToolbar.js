import React from 'react';
import PropTypes from 'prop-types';
import { Level, LevelItem } from '@patternfly/react-core';
import TopToolbar from '../Shared/top-toolbar';
import FilterToolbarItem from '../Shared/FilterToolbarItem';
import OrderToolbarItem from '../Shared/OrderToolbarItem';

const PlatformToolbar = ({ searchValue, onFilterChange }) => (
  <TopToolbar>
    <Level>
      <LevelItem>
        <FilterToolbarItem searchValue={ searchValue } onFilterChange={ onFilterChange } placeholder="Find a product" />
      </LevelItem>
      <LevelItem>
        <OrderToolbarItem />
      </LevelItem>
    </Level>
  </TopToolbar>
);

PlatformToolbar.propTypes = {
  searchValue: PropTypes.string,
  onFilterChange: PropTypes.func.isRequired
};

export default PlatformToolbar;
