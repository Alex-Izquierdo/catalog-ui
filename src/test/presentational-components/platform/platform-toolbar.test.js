import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { shallowToJson } from 'enzyme-to-json';
import PlatformToolbar from '../../../presentational-components/platform/platform-toolbar';
import { mockBreacrumbsStore } from '../../redux/redux-helpers';

describe('<PlatformToolbar />', () => {
  let initialProps;
  let Provider;
  beforeEach(() => {
    initialProps = {
      searchValue: '',
      onFilterChange: jest.fn(),
      title: 'toolbar title'
    };
    Provider = mockBreacrumbsStore();
  });

  it('should render correctly', () => {
    expect(shallowToJson(shallow(<PlatformToolbar { ...initialProps } />))).toMatchSnapshot();
  });

  it('should call search callback', () => {
    const onFilterChange = jest.fn();
    const wrapper = mount(
      <MemoryRouter initialEntries={ [ '/platforms' ] }>
        <Provider>
          <PlatformToolbar { ...initialProps } onFilterChange={ onFilterChange } />
        </Provider>
      </MemoryRouter>
    );
    wrapper.find('input').simulate('change', { target: { value: 'Foo' }});
    expect(onFilterChange).toHaveBeenLastCalledWith(expect.any(String), expect.objectContaining({ target: { value: 'Foo' }}));
  });
});
