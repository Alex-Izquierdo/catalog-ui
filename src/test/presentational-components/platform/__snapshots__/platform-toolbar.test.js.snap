// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformToolbar /> should render correctly 1`] = `
<TopToolbar
  paddingBottom={true}
>
  <TopToolbarTitle
    title="toolbar title"
  />
  <Level
    className=""
    gutter={null}
  >
    <Toolbar
      className={null}
    >
      <ToolbarGroup
        className={null}
      >
        <ToolbarItem
          className={null}
        >
          <FilterToolbarItem
            onFilterChange={[MockFunction]}
            placeholder="Filter by name..."
            searchValue=""
          />
        </ToolbarItem>
      </ToolbarGroup>
      <ToolbarGroup
        className={null}
      >
        <ToolbarItem
          className={null}
        />
      </ToolbarGroup>
    </Toolbar>
  </Level>
</TopToolbar>
`;
